receiver.go

```diff
@@ -45,16 +45,13 @@ func (r *Receiver) HandleMessage(ctx context.Context, handle func(*Message) erro
 	callHandler := func(msg *Message) error {
 		debug(3, "Receive() blocking %d", msg.deliveryID)
 		msg.receiver = r
-		if msg.doneSignal == nil {
-			msg.doneSignal = make(chan struct{})
-		}
-		go trackCompletion(msg)
-		// accept spontaneously when receiver is in ModeFirst
+		// we only need to track message disposition for mode second
 		// spec : http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#type-receiver-settle-mode
-		if r.link.receiverSettleMode.value() == ModeFirst {
-			if err := msg.Accept(ctx); err != nil {
-				return err
+		if r.link.receiverSettleMode.value() == ModeSecond {
+			if msg.doneSignal == nil {
```

Comment:
in that case we can move this initialization inside the trackCompletion func?

Reply:
Hmm yes I believe so, I will try this out tomorrow.

Reply:
Fixed